Added itemRef and itemId for complete microdata support.

btholt opened the pull request:
This is specifically addressing #2831. There are two properties for the microdata that the [WHATWG group](https://html.spec.whatwg.org/multipage/microdata.html#microdata-dom-api) identifies that the the [Schema.org](http://schema.org/docs/gs.html) doesn't identify: `itemRef` and `itemId`.

I struggled with how to case `itemId`: `itemId` (which is how the WHATWG cases it) or `itemID` (which is would be consistent with the internal casing to that JS file, as evidenced by `classID`. I opted for consistency with the WHATWG spec but I'm happy to change it if consistency inside the JS file is preferred.

zpao commented:
Thanks! I understand the struggle with `itemId`. I actually prefer `itemID` and think it's more idiomatic (I like to pretend `document.getElementById` is capitalized differently). Let's do that and be internally consistent. You *might* need to include a capitalization mapping lower down, but I don't think so (I can't quite remember, so if you could do a build and do a quick test to make sure changing a value across renders works, that would be awesome! We don't have a good system for automated property testing)

zpao commented on src/browser/ui/dom/HTMLDOMPropertyConfig.js:
```diff
@@ -164,11 +164,13 @@ var HTMLDOMPropertyConfig = {
     itemScope: MUST_USE_ATTRIBUTE | HAS_BOOLEAN_VALUE,
     itemType: MUST_USE_ATTRIBUTE,
+    itemId: MUST_USE_ATTRIBUTE,
+    itemRef: MUST_USE_ATTRIBUTE,
```
Could you actually move these up so we keep ABC order in this block?

Alternatively, we leave these two here and link to a separate doc where these 2 are mentioned.

btholt commented:
Changed it to `itemID` and manually tested it to make sure it was consistent across renders and it works fine. I also added another comment block to specify where those attributes are specified; good call!

zpao commented:
Awesome, thanks!
